plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -14,180 +14,51 @@
-                        .add(new ColumnMetadata("content", INTEGER))
-                        .add(new ColumnMetadata("file_path", VARCHAR))
-                        .add(new ColumnMetadata("file_format", VARCHAR))
-                        .add(new ColumnMetadata("record_count", BIGINT))
-                        .add(new ColumnMetadata("file_size_in_bytes", BIGINT))
-                        .add(new ColumnMetadata("column_sizes", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
-                        .add(new ColumnMetadata("value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
-                        .add(new ColumnMetadata("null_value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
-                        .add(new ColumnMetadata("nan_value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
-                        .add(new ColumnMetadata("lower_bounds", typeManager.getType(mapType(INTEGER.getTypeSignature(), VARCHAR.getTypeSignature()))))
-                        .add(new ColumnMetadata("upper_bounds", typeManager.getType(mapType(INTEGER.getTypeSignature(), VARCHAR.getTypeSignature()))))
-                        .add(new ColumnMetadata("key_metadata", VARBINARY))
-                        .add(new ColumnMetadata("split_offsets", new ArrayType(BIGINT)))
-                        .add(new ColumnMetadata("equality_ids", new ArrayType(INTEGER)))
-                        .build());
+        super(tableName, typeManager, icebergTable);
         this.snapshotId = requireNonNull(snapshotId, "snapshotId is null");
     }
 
-    @Override
-    public Distribution getDistribution()
-    {
-        return Distribution.SINGLE_COORDINATOR;
-    }
-
-    @Override
-    public ConnectorTableMetadata getTableMetadata()
-    {
-        return tableMetadata;
-    }
-
     @Override
     public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
     {
         if (snapshotId.isEmpty()) {
             return new FixedPageSource(ImmutableList.of());
         }
-        return new FixedPageSource(buildPages(tableMetadata, icebergTable, snapshotId.get()));
+        return super.pageSource(transactionHandle, session, constraint);
     }
 
-    private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, Table icebergTable, long snapshotId)
+    @Override
+    protected TableScan buildTableScan()
     {
-        PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
-        Map<Integer, Type> idToTypeMapping = getIcebergIdToTypeMapping(icebergTable.schema());
+        Table filesTable = createMetadataTableInstance(getIcebergTable(), FILES);
 
-        TableScan tableScan = icebergTable.newScan()
-                .useSnapshot(snapshotId)
+        TableScan tableScan = filesTable
```

Comment:
This change prevents accessing `$files` system table if delete-files exists. Such failure didn't happen before this commit. You can reproduce by below steps:
1. Start `IcebergQueryRunner`
2. Execute `DELETE FROM region WHERE regionkey = 1`
3. Execute `SELECT * FROM "region$files"`

```
class org.apache.iceberg.GenericDeleteFile cannot be cast to class org.apache.iceberg.DataFile (org.apache.iceberg.GenericDeleteFile and org.apache.iceberg.DataFile are in unnamed module of loader 'app')
```

Please add this test.

Reply:
@ebyhr I believe its ok to fail, but we should add a test-case?

Reply:
Can you explain why it's ok? Yes, we should add a test case.

Reply:
@ebyhr I didn't mean its ok, I was trying to ask question.

Reply:
added Delete statement in the tests.
Have to use the parent of  `DataFile` and `DeletedFile` that is `ContentFile`.